### ISSUE TITLE
[1.0] gatsby-sitemap-plugin failing on dev/build

### ISSUE THREAD

**Issue by okcoker** [gatsbyjs/gatsby#1397]
I'm using node 7.1.0.
After upgrading from 1.0.0-beta.5 to 1.0.5, I get the following error: 

```
Error: ./~/gatsby-plugin-sitemap/internals.js
Module not found: Error: Cannot resolve module 'fs' in /path/node_modules/gatsby-plugin-sitemap
resolve module fs in /path/node_modules/gatsby-plugin-sitemap
  looking for modules in /path/node_modules
    /path/node_modules/fs doesn't exist (module as directory)
    resolve 'file' fs in /path/node_modules
      resolve file
        /path/node_modules/fs doesn't exist
        /path/node_modules/fs.js doesn't exist
        /path/node_modules/fs.jsx doesn't exist
  looking for modules in /path/node_modules
    /path/node_modules/fs doesn't exist (module as directory)
    resolve 'file' fs in /path/node_modules
      resolve file
        /path/node_modules/fs doesn't exist
        /path/node_modules/fs.js doesn't exist
        /path/node_modules/fs.jsx doesn't exist
  looking for modules in /path/node_modules
    /path/node_modules/fs doesn't exist (module as directory)
    resolve 'file' fs in /path/node_modules
      resolve file
        /path/node_modules/fs doesn't exist
        /path/node_modules/fs.js doesn't exist
        /path/node_modules/fs.jsx doesn't exist
  looking for modules in /path/node_modules
    /path/node_modules/fs doesn't exist (module as directory)
    resolve 'file' fs in /path/node_modules
      resolve file
        /path/node_modules/fs doesn't exist
        /path/node_modules/fs.js doesn't exist
        /path/node_modules/fs.jsx doesn't exist
  looking for modules in /path/node_modules
    /path/node_modules/fs doesn't exist (module as directory)
    resolve 'file' fs in /path/node_modules
      resolve file
        /path/node_modules/fs doesn't exist
        /path/node_modules/fs.js doesn't exist
        /path/node_modules/fs.jsx doesn't exist
  looking for modules in /path/node_modules
    /path/node_modules/fs doesn't exist (module as directory)
    resolve 'file' fs in /path/node_modules
      resolve file
        /path/node_modules/fs doesn't exist
        /path/node_modules/fs.js doesn't exist
        /path/node_modules/fs.jsx doesn't exist
  looking for modules in /path/node_modules
    /path/node_modules/fs doesn't exist (module as directory)
    resolve 'file' fs in /path/node_modules
      resolve file
        /path/node_modules/fs doesn't exist
        /path/node_modules/fs.js doesn't exist
        /path/node_modules/fs.jsx doesn't exist
  looking for modules in /path/node_modules
    /path/node_modules/fs doesn't exist (module as directory)
    resolve 'file' fs in /path/node_modules
      resolve file
        /path/node_modules/fs doesn't exist
        /path/node_modules/fs.js doesn't exist
        /path/node_modules/fs.jsx doesn't exist
  looking for modules in /path/node_modules/gatsby/node_modules
    /path/node_modules/gatsby/node_modules/fs doesn't exist (module as directory)
    resolve 'file' fs in /path/node_modules/gatsby/node_modules
      resolve file
        /path/node_modules/gatsby/node_modules/fs doesn't exist
        /path/node_modules/gatsby/node_modules/fs.js doesn't exist
        /path/node_modules/gatsby/node_modules/fs.jsx doesn't exist
  looking for modules in /path/node_modules/gatsby/node_modules
    /path/node_modules/gatsby/node_modules/fs doesn't exist (module as directory)
    resolve 'file' fs in /path/node_modules/gatsby/node_modules
      resolve file
        /path/node_modules/gatsby/node_modules/fs doesn't exist
        /path/node_modules/gatsby/node_modules/fs.js doesn't exist
        /path/node_modules/gatsby/node_modules/fs.jsx doesn't exist
  looking for modules in /path/node_modules/gatsby/node_modules
    /path/node_modules/gatsby/node_modules/fs doesn't exist (module as directory)
    resolve 'file' fs in /path/node_modules/gatsby/node_modules
      resolve file
        /path/node_modules/gatsby/node_modules/fs doesn't exist
        /path/node_modules/gatsby/node_modules/fs.js doesn't exist
        /path/node_modules/gatsby/node_modules/fs.jsx doesn't exist
  looking for modules in /path/node_modules/gatsby/node_modules
    /path/node_modules/gatsby/node_modules/fs doesn't exist (module as directory)
    resolve 'file' fs in /path/node_modules/gatsby/node_modules
      resolve file
        /path/node_modules/gatsby/node_modules/fs doesn't exist
        /path/node_modules/gatsby/node_modules/fs.js doesn't exist
        /path/node_modules/gatsby/node_modules/fs.jsx doesn't exist
  looking for modules in /path/node_modules/gatsby/node_modules
    /path/node_modules/gatsby/node_modules/fs doesn't exist (module as directory)
    resolve 'file' fs in /path/node_modules/gatsby/node_modules
      resolve file
        /path/node_modules/gatsby/node_modules/fs doesn't exist
        /path/node_modules/gatsby/node_modules/fs.js doesn't exist
        /path/node_modules/gatsby/node_modules/fs.jsx doesn't exist
  looking for modules in /path/node_modules/gatsby/node_modules
    /path/node_modules/gatsby/node_modules/fs doesn't exist (module as directory)
    resolve 'file' fs in /path/node_modules/gatsby/node_modules
      resolve file
        /path/node_modules/gatsby/node_modules/fs doesn't exist
        /path/node_modules/gatsby/node_modules/fs.js doesn't exist
        /path/node_modules/gatsby/node_modules/fs.jsx doesn't exist
  looking for modules in /path/node_modules/gatsby/node_modules
    /path/node_modules/gatsby/node_modules/fs doesn't exist (module as directory)
    resolve 'file' fs in /path/node_modules/gatsby/node_modules
      resolve file
        /path/node_modules/gatsby/node_modules/fs doesn't exist
        /path/node_modules/gatsby/node_modules/fs.js doesn't exist
        /path/node_modules/gatsby/node_modules/fs.jsx doesn't exist
  looking for modules in /path/Sites/node_modules
    /path/Sites/node_modules/fs doesn't exist (module as directory)
    resolve 'file' fs in /path/Sites/node_modules
      resolve file
        /path/Sites/node_modules/fs doesn't exist
        /path/Sites/node_modules/fs.js doesn't exist
        /path/Sites/node_modules/fs.jsx doesn't exist
[/path/node_modules/fs]
[/path/node_modules/fs]
[/path/node_modules/fs]
[/path/node_modules/fs]
[/path/node_modules/fs]
[/path/node_modules/fs]
[/path/node_modules/fs]
[/path/node_modules/fs]
[/path/node_modules/fs]
[/path/node_modules/fs]
[/path/node_modules/fs]
[/path/node_modules/fs]
[/path/node_modules/fs]
[/path/node_modules/fs]
[/path/node_modules/fs]
[/path/node_modules/fs]
[/path/node_modules/fs.js]
[/path/node_modules/fs.js]
[/path/node_modules/fs.js]
[/path/node_modules/fs.js]
[/path/node_modules/fs.js]
[/path/node_modules/fs.js]
[/path/node_modules/fs.js]
[/path/node_modules/fs.js]
[/path/node_modules/fs.jsx]
[/path/node_modules/fs.jsx]
[/path/node_modules/fs.jsx]
[/path/node_modules/fs.jsx]
[/path/node_modules/fs.jsx]
[/path/node_modules/fs.jsx]
[/path/node_modules/fs.jsx]
[/path/node_modules/fs.jsx]
[/path/node_modules/gatsby/node_modules/fs]
[/path/node_modules/gatsby/node_modules/fs]
[/path/node_modules/gatsby/node_modules/fs]
[/path/node_modules/gatsby/node_modules/fs]
[/path/node_modules/gatsby/node_modules/fs]
[/path/node_modules/gatsby/node_modules/fs]
[/path/node_modules/gatsby/node_modules/fs]
[/path/node_modules/gatsby/node_modules/fs]
[/path/node_modules/gatsby/node_modules/fs]
[/path/node_modules/gatsby/node_modules/fs]
[/path/node_modules/gatsby/node_modules/fs]
[/path/node_modules/gatsby/node_modules/fs]
[/path/node_modules/gatsby/node_modules/fs]
[/path/node_modules/gatsby/node_modules/fs]
[/path/node_modules/gatsby/node_modules/fs.js]
[/path/node_modules/gatsby/node_modules/fs.js]
[/path/node_modules/gatsby/node_modules/fs.js]
[/path/node_modules/gatsby/node_modules/fs.js]
[/path/node_modules/gatsby/node_modules/fs.js]
[/path/node_modules/gatsby/node_modules/fs.js]
[/path/node_modules/gatsby/node_modules/fs.js]
[/path/node_modules/gatsby/node_modules/fs.jsx]
[/path/node_modules/gatsby/node_modules/fs.jsx]
[/path/node_modules/gatsby/node_modules/fs.jsx]
[/path/node_modules/gatsby/node_modules/fs.jsx]
[/path/node_modules/gatsby/node_modules/fs.jsx]
[/path/node_modules/gatsby/node_modules/fs.jsx]
[/path/node_modules/gatsby/node_modules/fs.jsx]
[/path/Sites/node_modules/fs]
[/path/Sites/node_modules/fs]
[/path/Sites/node_modules/fs.js]
[/path/Sites/node_modules/fs.jsx]
 @ ./~/gatsby-plugin-sitemap/internals.js 8:10-23,Error: Cannot find module "fs"
    at webpackMissingModule (render-page.js:38810:76)
    at Object.that (render-page.js:38810:155)
    at __webpack_require__ (render-page.js:30:30)
    at Object.defineProperty.value (render-page.js:38777:19)
    at __webpack_require__ (render-page.js:30:30)
    at Object.<anonymous> (render-page.js:38665:12)
    at __webpack_require__ (render-page.js:30:30)
    at Object.exports.__esModule (render-page.js:73:22)
    at __webpack_require__ (render-page.js:30:30)
    at render-page.js:50:18
```

**Comment by chiedo** [comment]
Getting the same

**Comment by chiedo** [comment]
Same goes with gatsby-plugin-feed.

**Comment by chiedo** [comment]
@KyleAMathews you mentioned it would be helpful to have tests written for the sitemap and feed plugin. What did you mean by that? I didn't see any tests in the gatsbygram folder.

**Comment by KyleAMathews** [comment]
@chiedo just what I said? Gatsbygram isn't a plugin so I'm not sure why you're comparing the two.

@nicholaswyoung any chance you could look into this soon?

**Comment by chiedo** [comment]
Hey @KyleAMathews sorry that wasn't clear!

I wasn't making a comparison. Every time I do a pull request a get an email/github comment saying `Deploy preview ready!` and pointing to a gatsbygram build. I assumed by test, you meant adding something to that project that could be verified...?

So let me rephrase.

How would I go about writing a test for both of those plugins (sitemap and feed) that could be automated with `yarn test` (Or would testing work some other way)?

I've looked at five different plugin folders, in the project root and in the documentation and didn't see anything about tests in any of those places except for `v0-tests` in the project root folder. If you could point me to an example for one of the tests written for one of the plugins, I can take it from there.

I don't see how you could write an automated test for these plugins without building a site and viewing it in the browser but in this context I'm sure you know something I don't.

**Comment by samzhao** [comment]
Just got the same problem with gatsby-plugin-feed,

```
Error: ./~/gatsby-plugin-feed/internals.js
Module not found: Error: Cannot resolve module 'fs' in /work/gatsby-site/node_modules/gatsby-plugin-feed
```
This was with the default gatsby blog starter boilerplate; I didn't add anything new to the generated files except for including the plugins and adding them to `gatsby-config.js`.

Any ideas?

**Comment by chiedo** [comment]
Been trying to debug @samzhao 

I was able to get the error to go away by changing this line
https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/utils/webpack.config.js#L523

But then other things broke. Hit a wall so far.

**Comment by samzhao** [comment]
@chiedo thanks for sharing!

The problem with that line is `("build-html" || "develop-html")` will always evaluate to `"build-html"` because it's just comparing between a non-empty string and another non-empty string — it's essentially `(true || true)`, so the first one is preferred.

In gatsby-plugin-feed, the stage is `"develop-html"`, so the statement `stage === ("build-html" || "stage-html")` evaluates to false, and the target ends up being `web` instead of `node`.

**Comment by KyleAMathews** [comment]
@chiedo search for folders named __tests__

**Comment by chiedo** [comment]
Sounds good. I'll run a find command.

I was being lazy lol 

Want me to start a new issue or continue the discussion here? If things aren't working as is and I can't figure out how to fix it, then naturally my tests will fail

**Comment by nicholaswyoung** [comment]
Looks like #1427 resolves this issue. Once a new Gatsby release is out, we can close this, do ya think @KyleAMathews?

**Comment by chiedo** [comment]
Awesome!! I knew I was close to finding a fix but couldn't quite find the flaw. Good team work

**Comment by KyleAMathews** [comment]
I made a new patch release — lemme know if things are working now.

**Comment by nicholaswyoung** [comment]
@KyleAMathews Looks like 1.0.7 resolves this issue. 👍

**Comment by chiedo** [comment]
Working like a charm!

**Comment by okcoker** [comment]
Working for me thanks!
